Theme: Add ordering to patterns.

StevenDufresne opened the pull request:
This PR adds the ability to sort patterns based on 2 dimensions:
- Newest
- Favorited

In the future, we hope to add support for `Featured` categories.

Fixes #161

## Needs Resolution

Currently, block patterns don't start with the meta key: `wporg-pattern-favorites` set. This means that when we run the sorting query it will only return the patterns that have favorites. That may not be the experience we are looking for. In order to correct it, we'll need to set a default value. Should we `register_post_meta` and set the default value to `0`? 

**Example**:
![](https://d.pr/i/fS8Jwd.gif)


### Screenshots
| Desktop | Mobile |
| --- | --- |
|  ![](https://d.pr/i/mg9zA8.png) | ![](https://d.pr/i/flLF4s.png) |

<!-- If your change has a visual component, add a screenshot here. A "before" screenshot would also be helpful. -->

### How to test the changes in this Pull Request:

1. Visit `/`
2. Use the dropdown
3. Expect that the results match

<!-- If you can, add the appropriate [Component] label(s). -->

ryelle commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-order-select/index.js:
```diff
@@ -0,0 +1,40 @@
+	}
+
+	return (
+		<div className="pattern-select-control">
```
This should use the same name as the component/folder, so either the class should be `pattern-order-select` or the component should be `pattern-select-control` (but that sounds like you're selecting a pattern).

StevenDufresne committed:
Add in a control to sort the patterns.

StevenDufresne committed:
Rename patter order select style.

StevenDufresne committed:
Rebuild the scss components base on renaming.

ryelle committed:
Rename component to "Navigation Layout"

- Since it can contain more than a menu, use "navigation"
- Use primary and secondary in place of left/right - it's not left/right on smaller screens or in RTL
- Move the menu & select styles into their CSS files
- Use primary/secondary div wrappers to make the layout CSS less dependent on specific content

ryelle committed:
Move the `nav` wrapper into the Menu component

ryelle committed:
Pass down a class name on the SkeletonWrapper

ryelle committed:
Fix select label spacing on RTL langauge

ryelle committed:
Move API adjustments to the plugin, with other API tweaks

ryelle committed:
Persist the ordering on pagination and between categories

StevenDufresne committed:
Fix the dropdown on mobile so it's not full width.
